src/Lykke.blue.Service.ReferralLinks/Controllers/ReferralLinksController.cs

```
@@ -0,0 +1,480 @@
+        }
+
+        /// <summary>
+        /// Request money transfer referral link - reserved for version 2   
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        //[HttpPost("giftCoinLinks")]
+        //[SwaggerOperation("RequestGiftCoinsReferralLink")]
+        //[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+        //[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.NotFound)]
+        //[ProducesResponseType(typeof(RequestRefLinkResponse), (int)HttpStatusCode.Created)]        
+        private async Task<IActionResult> RequestGiftCoinsReferralLink([FromBody] GiftCoinsReferralLinkRequest request)
+        {
+            if (request == null)
+            {
+                return await LogAndReturnBadRequest("", ControllerContext, Phrases.InvalidRequest);
+            }
+
+            if (request.Amount <= 0)
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidAmount);
+            }
+
+            if (String.IsNullOrEmpty(request.SenderClientId))
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidSenderClientId);
+            }
+
+            var asset = (await _assets.GetDictionaryAsync()).Values.FirstOrDefault(v => v.DisplayId == request.Asset);
+
+            if (String.IsNullOrEmpty(request.Asset) || asset == null)
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidAsset);
+            }
+
+            var clientBalances = await _balancesClient.GetClientBalances(request.SenderClientId);
+
+            if (clientBalances == null)
+            {
+                return await LogAndReturnNotFound(request, ControllerContext, $"Cant get clientBalance of asset {asset.Symbol} for client id {request.SenderClientId} from service {_settings.BalancesServiceClient.ServiceUrl}");
```

Even we don't need this code in this release, I think that using balances service url here is redundant. We can remove it from here, and change the using type of settings for this controller from "AppSettings" to "ReferralLinksSettings".

---

Are you suggesting to remove the check, create the link and then let it fail when we try to create the offchain transaction?
The idea behind this check is that we don't want to spam Firebase with unnecessary calls, when we know they will produce invalid links for sure.
Firebase has a licensing scheme that makes it expensive if you use it a lot.

Btw, we need it in the release, because we want to produce some Gift Links for a Christmas campaign.